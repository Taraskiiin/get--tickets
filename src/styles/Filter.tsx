import styled from 'styled-components';
import { ButtonGroup, Button } from 'react-bootstrap';

export const Nav = styled(ButtonGroup)`
  width: 100%;
  height: max-content;
  display: flex;
  justify-content: space-between;
  text-transform: uppercase;
  letter-spacing: 0.5px;
  border-radius: 10px;
  background-color: #fff;
  margin-bottom: 20px;
`;

export const StyledButton = styled(Button)`
  text-decoration: none;
  text-align: center;
  padding: 14px 0;
  font-size: 14px;
  color: #4a4a4a;
  background-color: #fff;
  border: 1px solid #333333;

  &:hover {
    color: #fff;
    background-color: #2196f3;
  }

  &:active {
    color: #fff;
    background-color: #2196f3;
  }
`;
