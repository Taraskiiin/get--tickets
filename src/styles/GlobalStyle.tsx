import { createGlobalStyle } from 'styled-components';

const backgroundColor = '#f3f7fa';

export const GlobalStyle = createGlobalStyle`
  * {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
  }
  body {
      background-color: ${backgroundColor};
  }
  .container{
      padding: 0 50px;
  }
  @media (max-width: 768px) {
      .container {
        padding: 0 10px;
      }
  }
  `;
