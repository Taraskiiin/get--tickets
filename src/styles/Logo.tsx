import styled from 'styled-components';

export const NavLink = styled.a`
  margin: 50px auto;
  width: 20%;
  img {
    max-width: 150px;
    height: 100%;
  }
  @media ${(props) => props.theme.device.tablet} {
    width: 35%;
  }
`;
