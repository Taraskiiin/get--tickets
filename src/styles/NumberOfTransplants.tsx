import styled from 'styled-components';
import { Form, Field } from 'formik';
import { Button } from 'react-bootstrap';

export const NumberOfTransplantsBlock = styled.div`
  height: max-content;
  display: flex;
  padding: 20px;
  flex-direction: column;
  justify-content: space-around;
  align-items: start;
  background-color: ${(props) => props.theme.colors.white};
  border-radius: ${(props) => props.theme.borderRadius};
  font-size: ${(props) => props.theme.fontSizes.m};
  margin-bottom: 20px;
  box-shadow: 3px 3px 2px 1px ${(props) => props.theme.colors.athensGray};

}
  @media ${(props) => props.theme.device.tablet} {
    font-size: ${(props) => props.theme.fontSizes.s};
    padding: 10px;
  }
  @media ${(props) => props.theme.device.mobileL} {
    font-size: ${(props) => props.theme.fontSizes.m};
    padding: 20px;
  }
`;

export const TitleNumberOfTransplantsBlock = styled.h3`
  font-size: ${(props) => props.theme.fontSizes.m};
  font-weight: normal;
  line-height: 10px;
  margin-bottom: 20px;
`;

export const FormStyled = styled(Form)`
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  > label {
    width: max-content;
  }
`;

export const CustomCheckbox = styled(Field)`
  -webkit-appearance: none;
  appearance: none;
  background-color: ${(props) => props.theme.colors.white};
  width: 14px;
  height: 14px;
  border: 0.5px solid ${(props) => props.theme.colors.dodgerBlue};
  cursor: pointer;
  margin-bottom: 5px;

  &::after {
    content: 'âœ“';
    color: ${(props) => props.theme.colors.dodgerBlue};
    margin-left: 1px;
    opacity: 0;
  }
  &:hover {
    background-color: ${(props) => props.theme.colors.catskillWhite};
  }
  &:checked:after {
    opacity: 1;
  }
`;

export const LabeltoCustomCheckbox = styled.label`
  cursor: pointer;
  color: ${(props) => props.theme.colors.codGray};
`;

export const SubmitBtn = styled(Button)`
height: max-content;
  max-width: 20px 
  text-align: center;
  border: none;
  background-color: ${(props) => props.theme.colors.whitecodGray};
  color: ${(props) => props.theme.colors.white};
  font-size: ${(props) => props.theme.fontSizes.s};
  text-transform: uppercase;
  
  @media ${(props) => props.theme.device.tablet} {
    max-width: 90%;
    padding: 2px 3px 2px 3px
  }
  @media ${(props) => props.theme.device.mobileL} {
    font-size: ${(props) => props.theme.fontSizes.m};
    padding: 5px 10px 5px 10px;
  }
`;
