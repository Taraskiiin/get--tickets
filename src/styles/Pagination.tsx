import styled from 'styled-components';
import { ButtonGroup, Button } from 'react-bootstrap';
import { NavLink } from 'react-router-dom';

export const PaginationList = styled(ButtonGroup)`
  height: 100%;
  max-width: 100%;
  display: flex;
  margin-bottom: 10px;
  box-shadow: 3px 3px 2px 1px ${(props) => props.theme.colors.athensGray};
  overflow: scroll;
`;

export const PaginationItem = styled(Button)`
  text-align: center;
  border: none;
  background-color: ${(props) => props.theme.colors.white};
  color: ${(props) => props.theme.colors.codGray};
  font-size: ${(props) => props.theme.fontSizes.l};

  &:active {
    color: ${(props) => props.theme.colors.white};
    background-color: ${(props) => props.theme.colors.dodgerBlue};
    box-shadow: 1px 4px ${(props) => props.theme.colors.athensGray};
  }
`;

export const Link = styled(NavLink)`
  text-decoration: none;
`;
