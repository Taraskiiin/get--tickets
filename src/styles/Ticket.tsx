import styled from 'styled-components';

export const Card = styled.div`
  background-color: #fff;
  height: max-content;
  border-radius: 10px;
  padding: 25px 50px 25px 20px;
  text-transform: uppercase;
  font-size: 12px;
  margin-bottom: 20px;
`;

export const Header = styled.div`
  display: grid;
  grid-template-columns: 2fr 1fr;
  align-items: center;
  font-size: 12px;
  margin-bottom: 10px;
`;

export const Content = styled.section`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
`;

export const ColumnInfo = styled.p`
  margin-bottom: 20px;
  > span {
    display: block;
  }
`;

export const ColumnInfoTitle = styled.span`
  color: #a0b0b9;
`;

export const ColumnInfoTime = styled.span`
  text-transform: lowercase;
`;

export const Price = styled.span`
  font-size: 18px;
  color: #2196f3;
`;
