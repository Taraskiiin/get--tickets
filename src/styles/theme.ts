export interface ITheme {
  colors: Record<string, string>;
  fontSizes: Record<string, string>;
  borderRadius: string;
  device: Record<string, string>;
}

export const theme: ITheme = {
  colors: {
    catskillWhite: '#f3f7fa',
    white: '#ffffff',
    dodgerBlue: '#2196f3',
    gullGray: '#A0B0B9',
    codGray: '#111111',
    athensGray: '#E5E9EC',
  },
  fontSizes: {
    l: '18px',
    m: '14px',
    s: '12px',
  },
  borderRadius: '10px',
  device: {
    mobileS: '(max-width: 320px)',
    mobileM: '(max-width: 375px)',
    mobileL: '(max-width: 425px)',
    tablet: '(max-width: 768px)',
    laptop: '(max-width: 1024px)',
    laptopL: '(max-width: 1440px)',
    desktop: '(max-width: 2560px)',
  },
};
