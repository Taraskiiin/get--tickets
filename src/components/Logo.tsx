import React from 'react';

import { NavLink } from '../styles/Logo';

import logo from '../assets/Logo.png';

const Logo = () => {
  return (
    <NavLink href="#">
      <img src={`${logo}`} alt="logo" />
    </NavLink>
  );
};

export default Logo;
