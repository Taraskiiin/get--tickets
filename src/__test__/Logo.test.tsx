import React from 'react';
import renderWith from './RenderWith';
import Logo from '../components/Logo';
import { theme } from '../styles/theme';

describe('<Logo /> testing', () => {
  it('Should render with ThemeProvider', () => {
    expect(renderWith.withTheme(<Logo />, theme)).toMatchSnapshot();
  });
});
