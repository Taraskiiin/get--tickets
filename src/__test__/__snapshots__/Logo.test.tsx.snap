// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> testing Should render with ThemeProvider 1`] = `
<a
  className="sc-bdvvtL hrGerV"
  href="#"
>
  <img
    alt="logo"
    src="undefined/Logo.png"
  />
</a>
`;
